tests/testGroupStudent.js

```diff
@@ -0,0 +1,856 @@
+const ERR = require('async-stacktrace');
+
+const config = require('../lib/config');
+const path = require('path');
+var assert = require('chai').assert;
+var request = require('request');
+var cheerio = require('cheerio');
+
+
+
+var sqldb = require('@prairielearn/prairielib/sql-db');
+var sqlLoader = require('@prairielearn/prairielib/sql-loader');
+var sql = sqlLoader.loadSqlEquiv(__filename);
+
+var helperServer = require('./helperServer');
+
+// cd /PrairieLearn
+// docker/start_postgres.sh
+// git config --global user.email "dev@illinois.edu"
+// git config --global user.name "Dev User"
+// npm test -- --grep "Group based homework assess control on student side"
+
+let page, elemList;
+const locals = {};
+locals.helperClient = require('./helperClient');
+locals.siteUrl = 'http://localhost:' + config.serverPort;
+locals.baseUrl = locals.siteUrl + '/pl';
+locals.courseInstanceUrl = locals.baseUrl + '/course_instance/1';
+locals.assessmentsUrl = locals.courseInstanceUrl + '/assessments';
+locals.courseDir = path.join(__dirname, '..', 'testCourse');
+
+const storedConfig = {};
+
+describe('Group based homework assess control on student side', function() {
+    this.timeout(20000);
+    before('set authenticated user', function(callback) {
+        storedConfig.authUid = config.authUid;
```

Comment:
same as above with the `storedConfig` &mdash; is this needed?

Reply:
Actually, when I trying to remove that part in testGroupStudent , it causes an error 403 in the testGroupScoreAndSync when entering the locals.instructorAssessmentsUrlGroupTab. When I remove that part in testGroupScoreAndSync, that does not affect the test. Could you help me to find why the change in that file might cause 403 error in testGroupScoreAndSync?

Reply:
This still needs to be understood and resolved.